scripts/Dockerfile

```diff
@@ -4,7 +4,5 @@ ARG TARGETARCH
 COPY envbuilder-${TARGETARCH} /.envbuilder/bin/envbuilder
 
 ENV KANIKO_DIR /.envbuilder
-# Kaniko looks for the Docker config at $DOCKER_CONFIG/config.json
-ENV DOCKER_CONFIG /.envbuilder
```

Comment:
This may seem controversial, but I think it definitely makes sense to set it in code if we allow dynamically changing `MagicDir`.